test/qd8-f16-qb4w-gemm-minmax.cc

```diff
@@ -0,0 +1,494 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// All rights reserved.
+//
+// Copyright 2019 Google LLC
+//
+// This source code is licensed under the BSD-style license found in the
+// LICENSE file in the root directory of this source tree.
+//
+// Auto-generated file. Do not edit!
+//   Specification: test/qd8-f16-qb4w-gemm-minmax.yaml
+//   Generator: tools/generate-gemm-test.py
+
+#include <cstddef>
+#include <functional>
+#include <string>
+#include <vector>
+
+#include <xnnpack/allocator.h>
+#include <xnnpack/common.h>
+#include <xnnpack/gemm.h>
+#include <xnnpack/igemm.h>
+#include <xnnpack/isa-checks.h>
+#include <xnnpack/microparams-init.h>
+#include <xnnpack/pack.h>
+#include <xnnpack/ppmm.h>
+#include <xnnpack/requantization.h>
+
+#include "gemm-microkernel-tester.h"
+#include <gtest/gtest.h>
+
+namespace {
+
+std::vector<GemmTestParams> CreateTests1(
+    size_t k_block, size_t adj_k_block,
+    size_t mr, size_t nr, size_t kr, size_t sr,
+    bool is_igemm,
+    std::function<void(GemmMicrokernelTester& tester)> test_func,
+    std::function<void()> isa_check = nullptr) {
+  std::string kbs = std::to_string(k_block);
+  std::string kb2s = std::to_string(k_block * 2);
+  std::string akbs = std::to_string(adj_k_block);
+  std::string nrs = std::to_string(nr);
+
+  std::vector<GemmTestParams> gemm_tests;
+  gemm_tests.reserve(42);
+
+  gemm_tests.push_back(GemmTestParams(
+      "k_eq_" + kbs,
+      GemmMicrokernelTester()
+          .mr(mr).nr(nr).kr(kr).sr(sr).m(mr).n(nr).k(k_block)
+          .b_zero_point(8)
+          .bl(kr * sr * 2)
```

Comment:
Nit - do codegen like,

const size_t bl = 2 * kr * sr;
```suggestion
          .bl(bl)
```

Reply:
I'm slightly hesitant to do this, just because the precedent for all of the other args is to do the math in-place. That being said, I'm happy to change it if you feel strongly.

Reply:
No I don't have a strong preference.